### PR TITLE
Fix possible encodings issues

### PR DESCRIPTION
**Pull request by trasher**
## Description

Fix possible encodings issues

### CODE REVIEW

File: src/Cartridge.php [hunk 1]
```diff
@@ -81,7 +81,7 @@ public static function showMassiveActionsSubForm(MassiveAction $ma)
                 $input['maxpages'] = isset($input['maxpages']) ? (int) $input['maxpages'] : '';
 
                 echo "<input type='text' name='pages' value=\"" . $input['maxpages'] . "\" size='6'>";
-                echo "<br><br>" . Html::submit(_x('button', 'Update'), ['name' => 'massiveaction']);
+                echo "<br><br>" . Html::submit(_sx('button', 'Update'), ['name' => 'massiveaction']);
```
Comment (cedric-anne) [inline src/Cartridge.php:84]:
```suggestion
                echo "<br><br>" . Html::submit(_x('button', 'Update'), ['name' => 'massiveaction']);
```

File: src/CommonDBVisible.php [hunk 2]
```diff
@@ -309,17 +309,17 @@ public function showVisibility()
                         Html::showMassiveActionCheckBox('Group_' . $this::getType(), $data["id"]);
                         echo "</td>";
                     }
-                    echo "<td>" . Group::getTypeName(1) . "</td>";
+                    echo "<td>" . htmlspecialchars(Group::getTypeName(1)) . "</td>";
 
                     $names   = Dropdown::getDropdownName('glpi_groups', $data['groups_id'], 1);
-                    $entname = sprintf(
+                    $entname = htmlspecialchars(sprintf(
```
Comment (cedric-anne) [inline src/CommonDBVisible.php:315]: The tooltip should not be escaped as it contains legitimate HTML code.
```suggestion
                    $entname = sprintf(
```

File: src/CommonDBVisible.php [hunk 3]
```diff
@@ -309,17 +309,17 @@ public function showVisibility()
...
                         __('%1$s %2$s'),
                         $names["name"],
```
Comment (cedric-anne) [inline src/CommonDBVisible.php:317]:
```suggestion
                        __s('%1$s %2$s'),
                        htmlspecialchars($names["name"]),
```

File: src/CommonDBVisible.php [hunk 4]
```diff
@@ -309,17 +309,17 @@ public function showVisibility()
...
                         Html::showToolTip($names["comment"], ['display' => false])
-                    );
+                    ));
```
Comment (cedric-anne) [inline src/CommonDBVisible.php:319]:
```suggestion
                    );
```

File: src/CommonDBVisible.php [hunk 5]
```diff
@@ -309,17 +309,17 @@ public function showVisibility()
...
                     if ($data['entities_id'] !== null) {
                         $entname = sprintf(
-                            __('%1$s / %2$s'),
+                            __s('%1$s / %2$s'),
                             $entname,
                             Dropdown::getDropdownName(
```
Comment (cedric-anne) [inline src/CommonDBVisible.php:324]: `Dropdown::getDropdownName()` returns a non encoded string, it should be escaped too.
```suggestion
                            htmlspecialchars(Dropdown::getDropdownName(
```

I cannot add a sugesstion for the closing parenthesis :/.

File: src/CommonDBVisible.php [hunk 6]
```diff
@@ -383,14 +383,14 @@ public function showVisibility()
...
                     $names   = Dropdown::getDropdownName('glpi_profiles', $data['profiles_id'], 1);
                     $tooltip = Html::showToolTip($names["comment"], ['display' => false]);
-                    $entname = sprintf(__('%1$s %2$s'), $names["name"], $tooltip);
+                    $entname = sprintf(__s('%1$s %2$s'), htmlspecialchars($names["name"]), $tooltip);
                     if ($data['entities_id'] !== null) {
                         $entname = sprintf(
-                            __('%1$s / %2$s'),
+                            __s('%1$s / %2$s'),
                             $entname,
                             Dropdown::getDropdownName(
```
Comment (cedric-anne) [inline src/CommonDBVisible.php:395]: Same here, it should be escaped.

File: src/CommonITILTask.php [hunk 7]
```diff
@@ -1944,20 +1944,21 @@ public static function showVeryShort($ID, $itemtype)
...
                    $item_link->getFormURLWithID($item_link->fields["id"]);
-            $link .= "&amp;forcetab=" . $tab_name . "$1";
+            $link .= "&amp;forcetab=" . htmlspecialchars($tab_name) . "$1";
             $link   .= "'>";
             $link    = sprintf(__('%1$s'), $link);
-            printf(
+            $link = sprintf(
                 __('%1$s %2$s'),
                 $link,
                 Html::resume_text(RichText::getTextFromHtml($job->fields['content'], false, true), 50)
```
Comment (cedric-anne) [inline src/CommonITILTask.php:1959]: Html::resume_text() outputs HTML code, so it should not be escaped again.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: src/CommonDBVisible.php (hunk 5 above)
Comment (trasher) [reply src/CommonDBVisible.php]: Fixed

File: src/CommonDBVisible.php (hunk 6 above)
Comment (trasher) [reply src/CommonDBVisible.php]: Fixed

File: src/CommonITILTask.php (hunk 7 above)
Comment (trasher) [reply src/CommonITILTask.php:1958]: Fixed

File: src/CommonITILTask.php [hunk 8]
```diff
@@ -1944,20 +1944,20 @@ public static function showVeryShort($ID, $itemtype)
...
             echo "</td>";
 
             echo "<td>";
-            $link = "<a id='" . strtolower($item_link->getType()) . "ticket" . $item_link->fields["id"] . $rand . "' href='" .
+            $link = "<a id='" . htmlspecialchars(strtolower($item_link->getType())) . "ticket" . htmlspecialchars($item_link->fields["id"] . $rand) . "' href='" .
                    $item_link->getFormURLWithID($item_link->fields["id"]);
-            $link .= "&amp;forcetab=" . $tab_name . "$1";
+            $link .= "&amp;forcetab=" . htmlspecialchars($tab_name) . "$1";
             $link   .= "'>";
-            $link    = sprintf(__('%1$s'), $link);
```
Comment (trasher) [inline src/CommonITILTask.php:1955]: Seems a bit useless

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**